api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -456,7 +459,35 @@ model NodePoolAutoScaling {
   @minValue(0)
   max?: int32;
 }
+/** Disk encryption settings for the OS and data disks */
+model Encryption {
+  /** Infrastructure-level encryption for the disks. Defaults to "disabled". */
+  infrastructureEncryption?: "enabled" | "disabled" | string = "disabled";
```

Comment:
That's interesting, I've not seen this style of defining string enums before.

Reply:
I think it works just to have disabled by default, but I can change it @mbarnes

Reply:
No I like it, it's more compact than the one-off string union types we've been doing.

I'll have to see what the generated JSON and Go code looks like, but if it's equivalent I'm tempted to apply this style to other existing enums like API visibility and outbound type.

Reply:
Just to close this thread, I decided _not_ to convert existing enums because it means we lose doc strings for the enum values.  (AFAIK there's no way to have doc strings for inline enum values.)  For super obvious cases like what you have here ("enabled" | "disabled") I think the inline syntax is fine.  Other enum values are less obvious and I think will benefit from having doc strings.